tests/integration/helpers.py

```diff
@@ -187,6 +191,43 @@ def initial_workload_is_ready(ops_test, app_names) -> bool:
     )
 
 
+def write_tester_alert_rule_file(rule, name):
+    """Inject a new alert rule into Prometheus Tester.
+
+    Args:
+        rule: a string containing Prometheus alert rule in YAML format.
+        name: a string name of alert rule file
+    """
+    rules_path = Path(TESTER_ALERT_RULES_PATH).joinpath(name)
+    with rules_path.open(mode="w") as f:
+        f.write(rule)
+
+
+def remove_tester_alert_rule_file(name):
```

Comment:
Can we avoid modifying the source tree at all?
If not, maybe add this to auto cleanup?

Reply:
I do not see how I can inject an alert rule through the mechanism provided by the Prometheus charm (`promtheus_alert_rules` folder) without "modifying the source" tree and without more complicated code logic.